curriculum/challenges/english/07-scientific-computing-with-python/learn-special-methods-by-building-a-vector-space/65f9b486989cb90ff3e77ac8.md

```diff
@@ -0,0 +1,65 @@
+---
+id: 65f9b486989cb90ff3e77ac8
+title: Step 34
+challengeType: 20
+dashedName: step-34
+---
+
+# --description--
+
+The `isinstance()` built-in function takes two arguments and returns a Boolean indicating if the object passed as the first argument is an instance of the class passed as the second argument.
```

Comment:
an instance or a subclass

we have the issue that:
```py
vR2 = R2Vector(x=1, y=1)
vR3 = R3Vector(x=1, y=1, z=1)
isinstance(vR2, R2Vector) # True
isinstance(vR2, R3Vector) # False
isinstance(vR3, R2Vector) # True !!!
isinstance(vR3, R3Vector) # True
```

Reply:
This isn't really wrong from the Python inheritance perspective.

Reply:
Not wrong by the Python inheritance perspective, but it is leads to wrong results with the classes structured in this way.